modules/core/src/v2/coinFactory.ts

```diff
@@ -68,9 +69,22 @@ import { Bcha } from './coins/bcha';
 import { Tbcha } from './coins/tbcha';
 import { Dot } from './coins/dot';
 import { Tdot } from './coins/tdot';
+import { Environments } from './environments';
 
 export type CoinConstructor = (bitgo: BitGo, staticsCoin?: Readonly<StaticsBaseCoin>) => BaseCoin;
 
+/**
+ * It creates a constructor that checks network env to choose.
+ * Avalanche network has the same contract address for Mainnet and Testnet.
+ * The issue is select the right one see with network is used.
+ */
```

Comment:
Please take another pass at this comment. I have no idea what this is trying to say.

Reply:
done, now is it  understand?